content/docs/setup/kubernetes/istio-cni-install/index.md

```diff
@@ -0,0 +1,175 @@
+
+| Hosted Cluster Type | Uses CNI | Required Non-default settings |
+|---------------------|----------|-------------------------------|
+| GKE 1.9+ default | N | |
+| GKE 1.9+ w/ [network-policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy) | Y | `istio-cni.cniBinDir=/home/kubernetes/bin` |
+| IKS (IBM cloud) | Y | |
+| EKS (AWS) | Y | |
+| AKS (Azure) | Y | |
+| Red Hat OpenShift 3.10+ | Y | |
+
+#### GKE Setup
+
+1.  Refer to the procedure to [prepare a GKE cluster for Istio](/docs/setup/kubernetes/platform-setup/gke/) and
+    enable [network-policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy) in your cluster.
+    *  Note for existing clusters this redeploys the nodes.
+
+1.  Install Istio via Helm including these options
+    `--set istio_cni.enabled=true --set istio-cni.cniBinDir=/home/kubernetes/bin`
+
+## Sidecar Injection Compatibility
+
+The use of the Istio CNI plugin requires Kubernetes pods to be deployed with a sidecar injection method
+that uses the `istio-sidecar-injector` configmap created from the Helm installation with the
+`istio_cni.enabled=true`.  Refer to [Istio sidecar injection](/docs/setup/kubernetes/sidecar-injection/)
+for details about Istio sidecar injection methods.
+
+The following sidecar injection methods are supported for use with the Istio CNI plugin:
+
+1.  [Automatic sidecar injection](/docs/setup/kubernetes/sidecar-injection/#automatic-sidecar-injection)
+1.  Manual sidecar injection with the `istio-sidecar-injector` configmap
+    1.  `istioctl kube-inject` using the configmap directly:
+
+        {{< text bash >}}
+        $ istioctl kube-inject -f deployment.yaml -o deployment-injected.yaml --injectConfigMapName istio-sidecar-injector
+        $ kubectl apply -f deployment-injected.yaml
+        {{< /text >}}
+
+    1.  `istioctl kube-inject` using a file created from the configmap:
+
+        {{< text bash >}}
+        $ kubectl -n istio-system get configmap istio-sidecar-injector -o=jsonpath='{.data.config}' > inject-config.yaml
+        $ istioctl kube-inject -f deployment.yaml -o deployment-injected.yaml --injectConfigFile inject-config.yaml
+        $ kubectl apply -f deployment-injected.yaml
+        {{< /text >}}
+
+## Operational Details
+
+The Istio CNI plugin handles Kubernetes pod create and delete events and does the following:
+
+1.  Identify Istio user application pods with Istio sidecars requiring traffic redirection
+1.  Perform pod network namespace configuration to redirect traffic to/from the Istio sidecar
+
+### Identifying pods requiring traffic redirection
+
+The Istio CNI plugin identifies pods requiring traffic redirection to/from the
+accompanying Istio proxy sidecar by checking that the pod meets all of the following conditions:
+
+1.  The pod is NOT in a Kubernetes namespace in the configured `exclude_namespaces` list
+1.  The pod has a container named `istio-proxy`
+1.  The pod has more than 1 container
```

Comment:
this is a little confusing.  Wouldn't this always be the case if line 140 was met?

Reply:
Not necessarily... e.g. `istio-proxy` is the container name for the Istio ingress and egress gateway pods--which have only a single container.